### PR TITLE
[DSI-7245, DSI-7250] Update DSI User's associated Entra account

### PR DESCRIPTION
**Pull request by trickyward**
### Summary
Jira: [DSI-7245](https://dfe-secureaccess.atlassian.net/browse/DSI-7245) - update name details in Entra
Jira: [DSI-7250](https://dfe-secureaccess.atlassian.net/browse/DSI-7250) - update email in Entra

### Changes
- Added dependency on `login.dfe.entra-auth-extensions` to access Graph API
- Resolved some lint warnings
- Modify `src/app/user/api/patchUser.js` so that if the user is entra linked
  - Call Entra if either the name or email properties are changed
  - If a call to Entra fails, then revert the changes to name or email and return error
- Add applicable unit tests for the above

### Note
- patchUser is called with either a name change or an email change, but not both.

### CODE REVIEW

**Review by ipredoi — APPROVED** [review]
Looks good 👍

File: src/app/user/api/patchUser.js [hunk 1]
```diff
@@ -39,19 +49,81 @@ const patchUser = async (req, res) => {
...
-      connectionString: config.notifications.connectionString,
-    });
-    const jobId = await serviceNotificationsClient.notifyUserUpdated(safeUser(updatedUser));
-    logger.info(`Send user updated notification for ${user.sub} with job id ${jobId} (reason: patch)`, { correlationId });
+  if (!!user.is_entra && !!user.entra_oid) {
+    let nameChangeFailed = false;
+    let emailChangeFailed = false;
+    const errorMessage = { type: 'error', message: 'Unable to update user record, there was an error with Entra' };
+
+    if (nameDetailsChanged === true) {
```
Comment (kruncher) [inline src/app/user/api/patchUser.js:73]: My eyes keep mixing up the naming `nameChangeFailed` and `nameDetailsChanged`; what you have makes complete sense but for some reason I keep getting tripped on this naming.

**Review by kruncher — DISMISSED** [review]
Seems good 👍

**Review by ipredoi — DISMISSED** [review]
Looks good

File: src/app/user/api/patchUser.js (hunk 1 above)
Comment (trickyward) [reply src/app/user/api/patchUser.js:73]: I can change the variable name(s) if you'd like?

**Review by shravani-kondaparti — APPROVED** [review]

**Pull request merged**